jina/drivers/cache.py

```diff
@@ -22,10 +27,21 @@ def __init__(self, with_serialization: bool = False, *args, **kwargs):
         """
         self.with_serialization = with_serialization
         super().__init__(*args, **kwargs)
+        self.field = kwargs['field'] or self.default_field
+        if self.field not in self.supported_fields:
+            raise ValueError(f"Field '{self.field}' not in supported list of {self.supported_fields}")
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
         for d in docs:
-            result = self.exec[d.id]
+            # TODO waiting for new API from docId refactor PR
+            if self.field == 'id':
+                result = self.exec[d.id]
+            elif self.field == 'content':
+                # TODO what about blob?
```

Comment:
blob is one of the potential contents

Reply:
yes but it's a different field, it's `.blob` So should that be a different field for the driver parameter? Because `.text` is the same as `.content`.